### PR TITLE
Add invocation count measurers example

### PR DESCRIPTION
**Pull request by kjanosz**
Standalone sbt example that demonstrates usage of two invocation count measurers.

### CODE REVIEW

**Comment by kjanosz** [comment]
Btw, forcing user to specify `MethodMatcher.MethodName("<init>")` in order to measurer non-standard class allocations looks a little bit clumsy, so maybe introducing some `MethodMatcher.Allocation` would be helpful?

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,8 @@ env:
     TASKS="bench:compile"
   - TEST_DIR=offline-regression-report
     TASKS="bench:compile"
+  - TEST_DIR=invocation-count-measurers
+    TASKS="bench:compile"
```
Comment (axel22) [inline .travis.yml:15]: Perhaps we should change these to `bench:test`.

Comment (kjanosz) [inline .travis.yml:15]: Was thinking about the same. Do you prefer to do it in a separate PR or in this one?

File: invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        exec.independentSamples -> 1 // we rather want one JVM instance since this measurer is deterministic
+      ) in {
+        l => l.map(Some(_))
+      }
+    }
+  }
+
+  // we want to count allocations of `Option` method invocations
+  override def measurer: Measurer = Measurer.MethodInvocationCount(
+    InvocationCountMatcher(ClassMatcher.Descendants(classOf[Option[_]], direct = true, withSelf = false), MethodMatcher.MethodName("<init>"))
```
Comment (axel22) [inline invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala]: Please break this into two lines for readability.

File: invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  val sizes = Gen.range("size")(300000, 1500000, 300000)
+
+  val lists = for {
+    size <- sizes
+  } yield (0 until size).toList
+
+  performance of "Option" in {
+    measure method "allocations" in {
+      using(lists) config (
+        exec.independentSamples -> 1 // we rather want one JVM instance since this measurer is deterministic
```
Comment (axel22) [inline invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala]: In this case, you could do this by overriding the method on the `PerformanceTest`.

File: .travis.yml (hunk 1 above)
Comment (axel22) [inline .travis.yml:15]: Whatever is more convenient for you. Since it will increase the running time, might be better to do a separate PR.

**Comment by axel22** [comment]
Awesome - LGTM!

File: invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala (hunk 3 above)
Comment (kjanosz) [inline invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala]: Could you clarify, because I'm not sure if you mean that I could override `exec.independentSamples` higher (e.g. in the `performance` line) or something else?

Comment (kjanosz) [inline invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala]: Btw, I've noticed also that setting:

```
exec.independentSamples -> 1
```

with e.g. `OfflineRegressionReport` throws an exception:

```
org.apache.commons.math3.exception.NotStrictlyPositiveException: degrees of freedom (0)
    at org.apache.commons.math3.distribution.TDistribution.<init>(TDistribution.java:92)
    at org.apache.commons.math3.distribution.TDistribution.<init>(TDistribution.java:71)
    at org.apache.commons.math3.distribution.TDistribution.<init>(TDistribution.java:55)
    at org.scalameter.utils.Statistics$.org$scalameter$utils$Statistics$$qt(Statistics.scala:189)
    at org.scalameter.utils.Statistics$.confidenceInterval(Statistics.scala:42)
    at org.scalameter.utils.Statistics$Standard2WayTest.ci1(Statistics.scala:60)
    at org.scalameter.utils.Statistics$OverlapTest.ci1(Statistics.scala:105)
    at org.scalameter.utils.Statistics$OverlapTest.<init>(Statistics.scala:110)
    at org.scalameter.reporting.RegressionReporter$Tester$OverlapIntervals.confidenceInterval(RegressionReporter.scala:370)
    at org.scalameter.reporting.RegressionReporter$Tester$OverlapIntervals.multiple(RegressionReporter.scala:345)
    at org.scalameter.reporting.RegressionReporter$Tester$OverlapIntervals$$anonfun$34.apply(RegressionReporter.scala:360)
    at org.scalameter.reporting.RegressionReporter$Tester$OverlapIntervals$$anonfun$34.apply(RegressionReporter.scala:358)
...
```

It it supposed to do so or is it a bug?

Comment (axel22) [inline invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala]: The `defaultConfig` method should of the `PerformanceTest` should define the `Context` that every test group starts with. If you override that, all the tests should get the keys specified there.

Comment (axel22) [inline invocation-count-measurers/src/bench/scala/org/scalameter/examples/ListBenchmark.scala]: About the exception, I had a feeling that we'd need to fix this. The regression reporter does some statistics to determine if the performance measurement is a regression, and it is undefined for 1 measurement.

This means that, in the `Measurer`, we should duplicate the measurement `N` times for deterministic tests, instead of just returning a sequence with one measurement.
